addons/sale_management/migrations/12.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2019 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_column_renames = {
+    'sale_quote_template': [
+        ('require_payment', None),
+    ],
+    'sale_quote_option': [
+        ('layout_category_id', None),  # just in case, we save unused value
+    ],
+}
+
+_field_renames = [
+    ('sale.order', 'sale_order', 'template_id', 'sale_order_template_id'),
+    ('sale.quote.line', 'sale_quote_line', 'quote_id',
+     'sale_order_template_id'),
+    ('sale.quote.option', 'sale_quote_option', 'template_id',
+     'sale_order_template_id'),
+    ('sale.order.line', 'sale_order_line', 'option_line_id',
+     'sale_order_option_ids'),
+    ('sale.quote.template', 'sale_quote_template', 'options',
+     'sale_order_template_option_ids'),
+]
+
+_model_renames = [
+    ('sale.quote.line', 'sale.order.template.line'),
+    ('sale.quote.option', 'sale.order.template.option'),
+    ('sale.quote.template', 'sale.order.template'),
+]
+
+_table_renames = [
+    ('sale_quote_line', 'sale_order_template_line'),
+    ('sale_quote_option', 'sale_order_template_option'),
+    ('sale_quote_template', 'sale_order_template'),
+]
+
+
+_white_list_fields = {
+    'sale.quote.template': (
+        "name", "sale_order_template_line_ids", "note",
+        "sale_order_template_option_ids", "number_of_days",
+        "require_signature", "require_payment", "mail_template_id", "active",
+        "id", "display_name", "name", "create_uid", "create_date", "write_uid",
+        "write_date", "__last_update"),
+    'sale.quote.line': (
+        "sequence", "sale_order_template_id", "name", "product_id",
+        "price_unit", "discount", "product_uom_qty", "product_uom_id",
+        "display_type", "id", "display_name", "create_uid", "create_date",
+        "write_uid", "write_date", "__last_update"),
+    'sale.quote.option': (
+        "sale_order_template_id", "product_id", "price_unit", "discount",
+        "uom_id", "quantity", "id", "display_name", "create_uid",
+        "create_date", "write_uid", "write_date", "__last_update"),
+}
+
+
+def put_in_correct_module(cr, white_list):
+    # avoid ir_model_data_module_name_uniq_index error
+    # because the module of this white fields were sale_quotation_builder
+    for model in white_list:
+        openupgrade.update_module_moved_fields(
+            cr, model, _white_list_fields[model],
```

Comment:
Use here `white_list` for coherence

Reply:
done. sorry, I forgot the thing about the authorship